src/app/utils/query-url-normalization.ts

```diff
@@ -0,0 +1,136 @@
+// Matches patterns like "('<1f7ff346-c174-45e5-af38-294e51d9969a>')" or "('<key>')"
+const keyIdRegex = /\(\'\<?\{?[ ?0-9a-zA-Z-]*\}?\>?\'\)/g;
+
+// Matches patterns like "(query=<key>)" or "(itemat=<key>,mode=<mode>)"
+const functionParamInitialRegex = /(?<=\=).*?(?=,)/g;
+const functionParamFinalRegex = /(?<=\=).[^,]*(?=\))/g;
+
+// Matches pattterns within quotes e.g "displayName: Gupta"
+const quotedTextRegex = /"([^"]*)"/g;
+
+// Matches PII patterns
+const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
+const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
+const guidRegex =
+ /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
+
+
+export function normalizeQueryUrl(incomingUrl: string) {
+  incomingUrl = decodeURIComponent(incomingUrl);
+
+  // Extract query string
+  let queryString: string = '';
+  const urlParts = incomingUrl.split('?');
+  if (urlParts.length > 1) {
+    const startOfQueryString = incomingUrl.indexOf('?');
+    queryString = incomingUrl.substring(startOfQueryString, incomingUrl.length - startOfQueryString);
+    queryString = processQueryParams(queryString.slice(1));
+  }
```

Comment:
A reusable function created to get the parts of the url exists as `parseSampleUrl` in the `sample-url=generation.ts` file.